### ISSUE TITLE
Puppet-lint 2.4.0 throws misleading warning on double-quoted strings with escaped variables

### ISSUE THREAD

**Issue by jlusiardi** [rodjek/puppet-lint#886]
Hi,

we have the following line of puppet code:
`'httpd.mod_jk.cluster.hosts'      => "\${host.name}"` and puppet-lint 2.4.0 emit this warning:
`WARNING: top-scope variable being used without an explicit namespace on line XXX`.

Puppet 2.3.0 did not complain at all. 

We fixed the warning by going to ' instead of ":
`'httpd.mod_jk.cluster.hosts'      => '\${host.name}'`
This seams correct to me, because by escaping the $, there is no variable in the string any more.

I guess the error message should be improved/corrected to:
`double quoted string containing no variables` (see https://github.com/rodjek/puppet-lint/blob/master/lib/puppet-lint/plugins/check_strings/double_quoted_strings.rb)

Regards
Joachim

**Comment by rodjek** [comment]
@jlusiardi Can you provide a more complete example of the manifest that reproduces the problem?

**Comment by jlusiardi** [comment]
If still of use:
```
class role::test {
    file {'/tmp/foo1':
        content => "\${foo}";
    }
    file {'/tmp/foo2':
        content => '\${foo}';
    }
}
```

No Warnings generated with 2.4.1. 
Warning with 2.4.0 was `WARNING: top-scope variable being used without an explicit namespace on line XXX`,

I consider this fixed! Thank you!

**Comment by rodjek** [comment]
> Should line content => "\${foo}"; issue a warning about a double quoted string without variable?

No, this is expected behaviour as you have an escaped character in the double quoted string which is a valid reason for using a double quoted string without any variable interpolation.

**Comment by jlusiardi** [comment]
Ok, thank you!
